src/core/reducers/categories.js

```diff
@@ -1,42 +1,123 @@
+/* @flow */
 import { oneLine } from 'common-tags';
 import config from 'config';
+import invariant from 'invariant';
 
-import {
-  ADDON_TYPE_THEME,
-  CATEGORIES_FETCH,
-  CATEGORIES_LOAD,
-  validAddonTypes,
-} from 'core/constants';
+import { ADDON_TYPE_THEME, validAddonTypes } from 'core/constants';
 import log from 'core/logger';
 
-export function emptyCategoryList() {
+export const FETCH_CATEGORIES: 'FETCH_CATEGORIES' = 'FETCH_CATEGORIES';
+export const LOAD_CATEGORIES: 'LOAD_CATEGORIES' = 'LOAD_CATEGORIES';
+
+// See: https://addons-server.readthedocs.io/en/latest/topics/api/categories.html#category-list
+type ExternalCategory = {|
+  id: string,
+  name: string,
+  slug: string,
+  application: string,
+  misc: boolean,
+  type: string,
+  weight: number,
+  description: string | null,
+|};
+
+// The absence of strict types is wanted because the logic in the reducer is a
+
+type CategoryMapType = {
+  [appName: string]: {
+    [addonType: string]: {
+      [addonSlug: string]: ExternalCategory,
+    },
+  },
+};
+
+export type CategoriesState = {|
+  categories: null | CategoryMapType,
+  loading: boolean,
+|};
+
+export const initialState: CategoriesState = {
+  categories: null,
+  loading: false,
+};
+
+type FetchCategoriesParams = {|
+  errorHandlerId: string,
+|};
+
+export type FetchCategoriesAction = {|
+  type: typeof FETCH_CATEGORIES,
+  payload: FetchCategoriesParams,
+|};
+
+export function fetchCategories({
+  errorHandlerId,
+}: FetchCategoriesParams): FetchCategoriesAction {
+  invariant(errorHandlerId, 'errorHandlerId is required');
+
+  return {
+    type: FETCH_CATEGORIES,
+    payload: { errorHandlerId },
+  };
+}
+
+type CategoriesLoadParams = {|
```

Comment:
You missed renaming this one to `LoadCategoriesParams`.

Reply:
oh :(